contrib/migrate/migrate.c

```diff
@@ -0,0 +1,897 @@
+/*
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include "migrate.h"
+
+#define VERSION_INFO 		72
+#define DATABASES_ENVNAME 	"CUBRID_DATABASES"
+#define DATABASES_FILENAME 	"databases.txt"
+#define PATH_MAX		256
+#define MAX_LINE		4096
+#define ENV_VAR_MAX		255
+
+static const char *catalog_query[] = {
+  /* alter catalog to add column */
+  "alter table _db_class add column unique_name varchar (255) after [class_of]",
+  "delete from _db_attribute where class_of.class_name = '_db_class' and rownum % 2 = 1",
+
+  "alter table db_serial add column unique_name varchar first",
+  "delete from _db_attribute where class_of.class_name = 'db_serial' and rownum % 2 = 1 and attr_name <> 'unique_name'",
+
+  "alter table db_trigger add column unique_name varchar after owner",
+  "delete from _db_attribute where class_of.class_name = 'db_trigger' and rownum % 2 = 1 and attr_name <> 'unique_name'",
+
+  /* alter catalog to modify _db_index_key */
+  "alter table _db_index_key modify column func varchar(1023)",
+  "delete from _db_attribute where class_of.class_name = '_db_index_key' and rownum % 2 = 1",
+
+  /* alter catalog to add tables and views (_db_server, _db_synonym, db_server, db_synonym) */
+  "create table [_db_server] ( \
+  	[link_name] character varying(255) not null, \
+  	[host] character varying(255), \
+  	[port] integer, \
+  	[db_name] character varying(255), \
+  	[user_name] character varying(255), \
+  	[password] character varying(1073741823), \
+  	[properties] character varying(2048), \
+  	[owner] [db_user] not null, \
+  	[comment] character varying(1024), \
+  	constraint [pk__db_server_link_name_owner] primary key ([link_name], [owner]) \
+  ) dont_reuse_oid, collate utf8_bin",
```

Comment:
Create queries should remove "collate utf8_bin" at the end.
This is because the character set may be different in the user environment.